libcudacxx/include/cuda/__iterator/transform_iterator.h

```diff
@@ -394,7 +394,7 @@ class transform_iterator : public __transform_iterator_category_base<_Fn, _Iter>
   template <class _Iter2 = _Iter>
   [[nodiscard]] _CCCL_API friend constexpr auto
   operator-(const transform_iterator& __lhs, const transform_iterator& __rhs)
-    _CCCL_TRAILING_REQUIRES(difference_type)(::cuda::std::sized_sentinel_for<_Iter2, _Iter2>)
+    _CCCL_TRAILING_REQUIRES(difference_type)(::cuda::std::__has_random_access_traversal<_Iter2>)
```

Comment:
Let's see if I can without this. Will revert.

Reply:
@miscco this fix is definitely needed for CUDA iterators. See error for example here: https://github.com/NVIDIA/cccl/actions/runs/18305094920/job/52120490028?pr=6019#step:4:1251